### DISCUSSION TITLE
Config help please

### DISCUSSION THREAD

**Discussion by Moshny** [devcompl/TTController#176] [Configuration] [answered]
Good afternoon, I really ask your help to write the config, I tried it myself, but several times I got confused with the setting so that some of the coolers work from the GPU temperature

and i cant make 4 different static colors on each ring on riing quad

controller 1
cooler 1 GPU
cooler 2 GPU
cooler 3 GPU
cooler 4 GPU
cooler 5 GPU

controller 2 
cooler 1 GPU
cooler 2 GPU
cooler 3 GPU
cooler 4 GPU
cooler 5 GPU

controller 3
cooler 1 GPU
cooler 2 GPU
cooler 3 GPU
cooler 4 GPU
cooler 5 GPU

controller 4
cooler 1 GPU
cooler 2 CPU
cooler 3 CPU
cooler 4 CPU
cooler 5 CPU

controller 5
cooler 1 CPU
cooler 2 CPU

controller 6
cooler 1 CPU
cooler 2 CPU
cooler 4 CPU
cooler 5 CPU

Total 26 coolers
16 GPU and 10 CPU

```
Info
-------------------------------
>
>        OS: Windows 10 Enterprise 2009 [22458.1000.amd64fre.rs_prerelease.210910-1430]
>        Build: 0.11.0-beta+Branch.master.Sha.42768ba987b22e18ff8cdd114f754e5f7ce16d72
>
-------------------------------
Applications
-------------------------------
>
>
-------------------------------
HID
-------------------------------
>
>  [0x264a, 0x2260]: \\?\hid#vid_264a&pid_2260#b&ed8d306&1&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
>  [0x264a, 0x2263]: \\?\hid#vid_264a&pid_2263#a&3a2219fd&2&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
>  [0x264a, 0x2261]: \\?\hid#vid_264a&pid_2261#c&1227ace&2&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
>  [0x264a, 0x2264]: \\?\hid#vid_264a&pid_2264#b&23e6974a&3&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
>  [0x264a, 0x2262]: \\?\hid#vid_264a&pid_2262#d&5442bc4&5&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
>  [0x264a, 0x2265]: \\?\hid#vid_264a&pid_2265#c&37e1123c&4&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
>
-------------------------------
Controllers
-------------------------------
>
>  Name: Riing Quad
>        VendorId: 9802
>        ProductId: 8800
>        Version: 3.0.3
>        Ports:
>                1:
>                        Data: [PortId: 1, Speed: 50, Rpm: 974, Unknown: 255]
>                        Identifier: [9802, 8800, 1]
>                2:
>                        Data: [PortId: 2, Speed: 50, Rpm: 980, Unknown: 255]
>                        Identifier: [9802, 8800, 2]
>                3:
>                        Data: [PortId: 3, Speed: 50, Rpm: 954, Unknown: 255]
>                        Identifier: [9802, 8800, 3]
>                4:
>                        Data: [PortId: 4, Speed: 50, Rpm: 975, Unknown: 255]
>                        Identifier: [9802, 8800, 4]
>                5:
>                        Data: [PortId: 5, Speed: 50, Rpm: 1019, Unknown: 255]
>                        Identifier: [9802, 8800, 5]
>        Available effect types: PerLed
>  Name: Riing Quad
>        VendorId: 9802
>        ProductId: 8803
>        Version: 3.0.3
>        Ports:
>                1:
>                        Data: [PortId: 1, Speed: 50, Rpm: 958, Unknown: 255]
>                        Identifier: [9802, 8803, 1]
>                2:
>                        Data: [PortId: 2, Speed: 50, Rpm: 809, Unknown: 255]
>                        Identifier: [9802, 8803, 2]
>                3:
>                        Data: [PortId: 3, Speed: 50, Rpm: 840, Unknown: 255]
>                        Identifier: [9802, 8803, 3]
>                4:
>                        Data: [PortId: 4, Speed: 50, Rpm: 830, Unknown: 255]
>                        Identifier: [9802, 8803, 4]
>                5:
>                        Data: [PortId: 5, Speed: 50, Rpm: 838, Unknown: 255]
>                        Identifier: [9802, 8803, 5]
>        Available effect types: PerLed
>  Name: Riing Quad
>        VendorId: 9802
>        ProductId: 8801
>        Version: 3.0.3
>        Ports:
>                1:
>                        Data: [PortId: 1, Speed: 50, Rpm: 961, Unknown: 255]
>                        Identifier: [9802, 8801, 1]
>                2:
>                        Data: [PortId: 2, Speed: 50, Rpm: 999, Unknown: 255]
>                        Identifier: [9802, 8801, 2]
>                3:
>                        Data: [PortId: 3, Speed: 50, Rpm: 1003, Unknown: 255]
>                        Identifier: [9802, 8801, 3]
>                4:
>                        Data: [PortId: 4, Speed: 50, Rpm: 933, Unknown: 255]
>                        Identifier: [9802, 8801, 4]
>                5:
>                        Data: [PortId: 5, Speed: 50, Rpm: 980, Unknown: 255]
>                        Identifier: [9802, 8801, 5]
>        Available effect types: PerLed
>  Name: Riing Quad
>        VendorId: 9802
>        ProductId: 8804
>        Version: 3.0.3
>        Ports:
>                1:
>                        Data: [PortId: 1, Speed: 50, Rpm: 830, Unknown: 255]
>                        Identifier: [9802, 8804, 1]
>                2:
>                        Data: [PortId: 2, Speed: 50, Rpm: 829, Unknown: 255]
>                        Identifier: [9802, 8804, 2]
>                3:
>                        Data: [PortId: 3, Speed: 60, Rpm: 0, Unknown: 255]
>                        Identifier: [9802, 8804, 3]
>                4:
>                        Data: [PortId: 4, Speed: 60, Rpm: 0, Unknown: 255]
>                        Identifier: [9802, 8804, 4]
>                5:
>                        Data: [PortId: 5, Speed: 60, Rpm: 0, Unknown: 255]
>                        Identifier: [9802, 8804, 5]
>        Available effect types: PerLed
>  Name: Riing Quad
>        VendorId: 9802
>        ProductId: 8802
>        Version: 3.0.3
>        Ports:
>                1:
>                        Data: [PortId: 1, Speed: 50, Rpm: 968, Unknown: 255]
>                        Identifier: [9802, 8802, 1]
>                2:
>                        Data: [PortId: 2, Speed: 50, Rpm: 973, Unknown: 255]
>                        Identifier: [9802, 8802, 2]
>                3:
>                        Data: [PortId: 3, Speed: 50, Rpm: 995, Unknown: 255]
>                        Identifier: [9802, 8802, 3]
>                4:
>                        Data: [PortId: 4, Speed: 50, Rpm: 945, Unknown: 255]
>                        Identifier: [9802, 8802, 4]
>                5:
>                        Data: [PortId: 5, Speed: 50, Rpm: 960, Unknown: 255]
>                        Identifier: [9802, 8802, 5]
>        Available effect types: PerLed
>  Name: Riing Quad
>        VendorId: 9802
>        ProductId: 8805
>        Version: 3.0.3
>        Ports:
>                1:
>                        Data: [PortId: 1, Speed: 50, Rpm: 881, Unknown: 255]
>                        Identifier: [9802, 8805, 1]
>                2:
>                        Data: [PortId: 2, Speed: 50, Rpm: 839, Unknown: 255]
>                        Identifier: [9802, 8805, 2]
>                3:
>                        Data: [PortId: 3, Speed: 60, Rpm: 0, Unknown: 255]
>                        Identifier: [9802, 8805, 3]
>                4:
>                        Data: [PortId: 4, Speed: 50, Rpm: 987, Unknown: 255]
>                        Identifier: [9802, 8805, 4]
>                5:
>                        Data: [PortId: 5, Speed: 50, Rpm: 980, Unknown: 255]
>                        Identifier: [9802, 8805, 5]
>        Available effect types: PerLed
>
-------------------------------
Sensors
-------------------------------
>
>  Nuvoton NCT6798D:
>        Temperature:
>                Temperature #1 (/lpc/nct6798d/temperature/1): 42.5 °C
>                Temperature #2 (/lpc/nct6798d/temperature/2): 37.0 °C
>                Temperature #3 (/lpc/nct6798d/temperature/3): 22.0 °C
>                Temperature #5 (/lpc/nct6798d/temperature/5): 103.0 °C
>                Temperature #6 (/lpc/nct6798d/temperature/6): 31.0 °C
>
>  AMD Ryzen 9 5950X 1:
>        Temperature:
>                Core (Tctl/Tdie) (/amdcpu/0/temperature/2): 57.6 °C
>                CCD1 (Tdie) (/amdcpu/0/temperature/3): 60.8 °C
>                CCD2 (Tdie) (/amdcpu/0/temperature/4): 45.5 °C
>                CCDs Max (Tdie) (/amdcpu/0/temperature/5): 60.8 °C
>                CCDs Average (Tdie) (/amdcpu/0/temperature/6): 53.1 °C
>
>  NVIDIA NVIDIA GeForce RTX 3090:
>        Temperature:
>                GPU Core (/gpu-nvidia/0/temperature/0): 30.0 °C
>
>  NVIDIA NVIDIA GeForce RTX 3090:
>        Temperature:
>                GPU Core (/gpu-nvidia/1/temperature/0): 30.0 °C
>
>  Samsung SSD 980 PRO 2TB:
>        Temperature:
>                Temperature (/nvme/0/temperature/0): 0.0 °C
>
-------------------------------
Press any key to continue...
```

**Comment by MoshiMoshi0** [comment, marked as answer]
Damn, that many fans will be a nice test for TTController. 
If I didnt forget anything the config below should work.

You can tweak: 
* "CurvePoints" for the GPU and CPU profile in "temperature, speed" format
* Static color values for GPU and CPU profile in "red, green, blue" format. Zones are currently set to 1=red, 2=green, 3=blue, 4=white
* Color and/or speed for the "shutdown" profile that is applied when you turn of the pc or the service
* "CriticalValue" for the sensors, I set it to 90*C, if any sensor reaches that temperature all fans go to 100%

Also thanks for the donation :)

```json
{
  "Profiles": [
    {
      "Name": "GPU",
      "Ports": [
        [9802, 8800, 1], [9802, 8800, 2], [9802, 8800, 3], [9802, 8800, 4], [9802, 8800, 5],
        [9802, 8801, 1], [9802, 8801, 2], [9802, 8801, 3], [9802, 8801, 4], [9802, 8801, 5],
        [9802, 8802, 1], [9802, 8802, 2], [9802, 8802, 3], [9802, 8802, 4], [9802, 8802, 5],
        [9802, 8803, 1]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [60, 20],
              [70, 25],
              [75, 40],
              [80, 65],
              [85, 100]
            ],
            "Sensors": ["/gpu-nvidia/0/temperature/0", "/gpu-nvidia/1/temperature/0"],
            "SensorMixFunction": "Maximum",
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "ZoneEffect",
          "Config":{
            "Effects": [
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [255, 0, 0] } }
              },
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [0, 255, 0] } }
              },
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [0, 0, 255] } }
              },
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [255, 255, 255] } }
              }
            ],
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "CPU",
      "Ports": [
        [9802, 8803, 2], [9802, 8803, 3], [9802, 8803, 4], [9802, 8803, 5],
        [9802, 8804, 1], [9802, 8804, 2],
        [9802, 8805, 1], [9802, 8805, 2], [9802, 8805, 4], [9802, 8805, 5]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [60, 20],
              [70, 25],
              [75, 40],
              [80, 65],
              [85, 100]
            ],
            "Sensors": ["/amdcpu/0/temperature/2"],
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "ZoneEffect",
          "Config":{
            "Effects": [
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [255, 0, 0] } }
              },
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [0, 255, 0] } }
              },
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [0, 0, 255] } }
              },
              {
                "Type": "StaticColorEffect",
                "Config": { "Color": { "Full": [255, 255, 255] } }
              }
            ],
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8800, 1], [9802, 8800, 2], [9802, 8800, 3], [9802, 8800, 4], [9802, 8800, 5],
        [9802, 8801, 1], [9802, 8801, 2], [9802, 8801, 3], [9802, 8801, 4], [9802, 8801, 5],
        [9802, 8802, 1], [9802, 8802, 2], [9802, 8802, 3], [9802, 8802, 4], [9802, 8802, 5],
        [9802, 8803, 1],
        [9802, 8803, 2], [9802, 8803, 3], [9802, 8803, 4], [9802, 8803, 5],
        [9802, 8804, 1], [9802, 8804, 2],
        [9802, 8805, 1], [9802, 8805, 2], [9802, 8805, 4], [9802, 8805, 5]
      ],
      "Speed": 35,
      "Color": {
        "Full": [255, 0, 0]
      }
    }
  ],
  "SensorConfigs": [
    {
      "Sensors": [
        "/intelcpu/0/temperature/8", 
        "/gpu-nvidia/0/temperature/0", 
        "/gpu-nvidia/1/temperature/0"
      ],
      "Config": {
        "CriticalValue": 90
      }
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8800, 1], [9802, 8800, 2], [9802, 8800, 3], [9802, 8800, 4], [9802, 8800, 5],
        [9802, 8801, 1], [9802, 8801, 2], [9802, 8801, 3], [9802, 8801, 4], [9802, 8801, 5],
        [9802, 8802, 1], [9802, 8802, 2], [9802, 8802, 3], [9802, 8802, 4], [9802, 8802, 5],
        [9802, 8803, 1],
        [9802, 8803, 2], [9802, 8803, 3], [9802, 8803, 4], [9802, 8803, 5],
        [9802, 8804, 1], [9802, 8804, 2],
        [9802, 8805, 1], [9802, 8805, 2], [9802, 8805, 4], [9802, 8805, 5]
      ],
      "Config": {
        "DeviceType": "RiingQuad",
        "IgnoreColorCache": true
      }
    }
  ]
}
```

**Reply by Moshny** [reply]
> Damn, that many fans will be a nice test for TTController.
> If I didnt forget anything the config below should work.
> 
> You can tweak:
> 
> * "CurvePoints" for the GPU and CPU profile in "temperature, speed" format
> * Static color values for GPU and CPU profile in "red, green, blue" format. Zones are currently set to 1=red, 2=green, 3=blue, 4=white
> * Color and/or speed for the "shutdown" profile that is applied when you turn of the pc or the service
> * "CriticalValue" for the sensors, I set it to 90*C, if any sensor reaches that temperature all fans go to 100%
> 
> Also thanks for the donation :)
> 
> ```json
> {
>   "Profiles": [
>     {
>       "Name": "GPU",
>       "Ports": [
>         [9802, 8800, 1], [9802, 8800, 2], [9802, 8800, 3], [9802, 8800, 4], [9802, 8800, 5],
>         [9802, 8801, 1], [9802, 8801, 2], [9802, 8801, 3], [9802, 8801, 4], [9802, 8801, 5],
>         [9802, 8802, 1], [9802, 8802, 2], [9802, 8802, 3], [9802, 8802, 4], [9802, 8802, 5],
>         [9802, 8803, 1]
>       ],
>       "SpeedControllers": [
>         {
>           "Type": "PwmSpeedController", 
>           "Config": {
>             "CurvePoints": [
>               [60, 20],
>               [70, 25],
>               [75, 40],
>               [80, 65],
>               [85, 100]
>             ],
>             "Sensors": ["/gpu-nvidia/0/temperature/0", "/gpu-nvidia/1/temperature/0"],
>             "SensorMixFunction": "Maximum",
>             "Trigger": {
>               "Type": "AlwaysTrigger"
>             }
>           }
>         }
>       ],
>       "Effects": [
>         {
>           "Type": "ZoneEffect",
>           "Config":{
>             "Effects": [
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [255, 0, 0] } }
>               },
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [0, 255, 0] } }
>               },
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [0, 0, 255] } }
>               },
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [255, 255, 255] } }
>               }
>             ],
>             "Trigger": {
>               "Type": "AlwaysTrigger"
>             }
>           }
>         }
>       ]
>     },
>     {
>       "Name": "CPU",
>       "Ports": [
>         [9802, 8803, 2], [9802, 8803, 3], [9802, 8803, 4], [9802, 8803, 5],
>         [9802, 8804, 1], [9802, 8804, 2],
>         [9802, 8805, 1], [9802, 8805, 2], [9802, 8805, 4], [9802, 8805, 5]
>       ],
>       "SpeedControllers": [
>         {
>           "Type": "PwmSpeedController", 
>           "Config": {
>             "CurvePoints": [
>               [60, 20],
>               [70, 25],
>               [75, 40],
>               [80, 65],
>               [85, 100]
>             ],
>             "Sensors": ["/amdcpu/0/temperature/2"],
>             "Trigger": {
>               "Type": "AlwaysTrigger"
>             }
>           }
>         }
>       ],
>       "Effects": [
>         {
>           "Type": "ZoneEffect",
>           "Config":{
>             "Effects": [
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [255, 0, 0] } }
>               },
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [0, 255, 0] } }
>               },
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [0, 0, 255] } }
>               },
>               {
>                 "Type": "StaticColorEffect",
>                 "Config": { "Color": { "Full": [255, 255, 255] } }
>               }
>             ],
>             "Trigger": {
>               "Type": "AlwaysTrigger"
>             }
>           }
>         }
>       ]
>     }
>   ],
>   "ComputerStateProfiles": [
>     {
>       "StateType": "Shutdown",
>       "Ports": [
>         [9802, 8800, 1], [9802, 8800, 2], [9802, 8800, 3], [9802, 8800, 4], [9802, 8800, 5],
>         [9802, 8801, 1], [9802, 8801, 2], [9802, 8801, 3], [9802, 8801, 4], [9802, 8801, 5],
>         [9802, 8802, 1], [9802, 8802, 2], [9802, 8802, 3], [9802, 8802, 4], [9802, 8802, 5],
>         [9802, 8803, 1],
>         [9802, 8803, 2], [9802, 8803, 3], [9802, 8803, 4], [9802, 8803, 5],
>         [9802, 8804, 1], [9802, 8804, 2],
>         [9802, 8805, 1], [9802, 8805, 2], [9802, 8805, 4], [9802, 8805, 5]
>       ],
>       "Speed": 35,
>       "Color": {
>         "Full": [255, 0, 0]
>       }
>     }
>   ],
>   "SensorConfigs": [
>     {
>       "Sensors": [
>         "/intelcpu/0/temperature/8", 
>         "/gpu-nvidia/0/temperature/0", 
>         "/gpu-nvidia/1/temperature/0"
>       ],
>       "Config": {
>         "CriticalValue": 90
>       }
>     }
>   ],
>   "PortConfigs": [
>     {
>       "Ports": [
>         [9802, 8800, 1], [9802, 8800, 2], [9802, 8800, 3], [9802, 8800, 4], [9802, 8800, 5],
>         [9802, 8801, 1], [9802, 8801, 2], [9802, 8801, 3], [9802, 8801, 4], [9802, 8801, 5],
>         [9802, 8802, 1], [9802, 8802, 2], [9802, 8802, 3], [9802, 8802, 4], [9802, 8802, 5],
>         [9802, 8803, 1],
>         [9802, 8803, 2], [9802, 8803, 3], [9802, 8803, 4], [9802, 8803, 5],
>         [9802, 8804, 1], [9802, 8804, 2],
>         [9802, 8805, 1], [9802, 8805, 2], [9802, 8805, 4], [9802, 8805, 5]
>       ],
>       "Config": {
>         "DeviceType": "RiingQuad",
>         "IgnoreColorCache": true
>       }
>     }
>   ]
> }
> ```

THANKS VERY MUCH, MY RESPECT! EVERYTHING WORKS :)

and such a question, how to make it all start together with the system and work without pressing buttons

**Reply by Moshny** [reply]
THANKS VERY MUCH, MY RESPECT!

and such a question, how to make it all start together with the system and work without pressing buttons

**Reply by MoshiMoshi0** [reply]
Installing and other stuff should be pretty easy to follow from the [documentation](https://moshimoshi0.github.io/TTController/#installing-the-service)

**Reply by Moshny** [reply]
> Installing and other stuff should be pretty easy to follow from the [documentation](https://moshimoshi0.github.io/TTController/#installing-the-service)

thanks, and sorry, in your config have       "Sensors": [
        "/intelcpu/0/temperature/8",  
        i have amd 5950x, all ok?

**Reply by MoshiMoshi0** [reply]
Oh, right, copied and didnt change. Yours should be `"/amdcpu/0/temperature/2"`.
